tensorboard/data_compat_test.py

```diff
@@ -201,10 +201,76 @@ def test_histogram(self):
         self.assertEqual(expected_metadata, new_value.metadata)
         self.assertTrue(new_value.HasField("tensor"))
         buckets = tensor_util.make_ndarray(new_value.tensor)
+        for bucket in buckets:
+            # No `backwards` buckets.
+            self.assertLessEqual(bucket[0], bucket[1])
         self.assertEqual(old_value.histo.min, buckets[0][0])
         self.assertEqual(old_value.histo.max, buckets[-1][1])
         self.assertEqual(23 * 45, buckets[:, 2].astype(int).sum())
 
+    def test_empty_histogram(self):
+        with tf.compat.v1.Graph().as_default():
+            old_op = tf.compat.v1.summary.histogram(
+                "empty_yet_important", tf.constant([])
+            )
+            old_value = self._value_from_op(old_op)
+        assert old_value.HasField("histo"), old_value
+        new_value = data_compat.migrate_value(old_value)
+
+        self.assertEqual("empty_yet_important", new_value.tag)
+        expected_metadata = histogram_metadata.create_summary_metadata(
+            display_name="empty_yet_important", description=""
+        )
+        self.assertEqual(expected_metadata, new_value.metadata)
+        self.assertTrue(new_value.HasField("tensor"))
+        buckets = tensor_util.make_ndarray(new_value.tensor)
+        self.assertEmpty(buckets)
+
+    def test_single_value_histogram(self):
+        with tf.compat.v1.Graph().as_default():
+            old_op = tf.compat.v1.summary.histogram(
+                "single_value_data", tf.constant([1] * 1024)
+            )
+            old_value = self._value_from_op(old_op)
+        assert old_value.HasField("histo"), old_value
+        new_value = data_compat.migrate_value(old_value)
+
+        self.assertEqual("single_value_data", new_value.tag)
+        expected_metadata = histogram_metadata.create_summary_metadata(
+            display_name="single_value_data", description=""
+        )
+        self.assertEqual(expected_metadata, new_value.metadata)
+        self.assertTrue(new_value.HasField("tensor"))
+        buckets = tensor_util.make_ndarray(new_value.tensor)
+        # Only one bucket is kept.
+        self.assertEqual((1, 3), buckets.shape)
+        self.assertEqual(1, buckets[0][0])
+        self.assertEqual(1, buckets[-1][1])
+        self.assertEqual(1024, buckets[0][2])
+
+    def test_histogram_with_empty_buckets_on_both_ends(self):
```

Comment:
It might be worth an additional test case using a histogram with data that has extremal values, e.g. [-1e20, 1e20], which should produce counts in the "farthest out" buckets that the legacy histogram format can generate.  In particular, in that case the final bucket in the legacy histogram format is non-empty (whereas usually it's empty).

Reply:
Added.